### PR TITLE
HDFS-15047. Document the new decommission monitor (HDFS-14854).

### PR DESCRIPTION
**Pull request by iwasakims**

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m 48s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  markdownlint  |   0m  1s |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  18m 30s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   1m  9s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  32m 17s |  branch has no errors when building and testing our client artifacts.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m  6s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   1m  4s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  13m 21s |  patch has no errors when building and testing our client artifacts.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 31s |  The patch does not generate ASF License warnings.  |
|  |   |  52m 51s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1755/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1755 |
| Optional Tests | dupname asflicense mvnsite markdownlint |
| uname | Linux 4906467b7026 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / c174d50 |
| Max. process+thread count | 412 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs U: hadoop-hdfs-project/hadoop-hdfs |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1755/1/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdfs-project/hadoop-hdfs/src/site/markdown/HdfsDataNodeAdminGuide.md [hunk 1]
```diff
@@ -148,6 +148,35 @@ dfs.namenode.decommission.blocks.per.interval
...
+------------
+
+Original decommissioning algorithm has issues when DataNodes having lots of
+blocks are decommissioned such as
+
+* Write lock in the NameNode could be held for a long time for queueing re-replication.
+* Re-replication work progresses node by node if there are multiple decommissioning DataNodes.
+
+[HDFS-14854](https://issues.apache.org/jira/browse/HDFS-14854) introduced
+new decommission monitor in order to migigates those issues.
```
Comment (goiri) [inline hadoop-hdfs-project/hadoop-hdfs/src/site/markdown/HdfsDataNodeAdminGuide.md]: "introduced *a* new decommission monitor in order to *mitigate* those issues."

File: hadoop-hdfs-project/hadoop-hdfs/src/site/markdown/HdfsDataNodeAdminGuide.md [hunk 2]
```diff
@@ -148,6 +148,35 @@ dfs.namenode.decommission.blocks.per.interval
 dfs.namenode.decommission.max.concurrent.tracked.nodes
 ```
 
+
+Backing-off Decommission Monitor (experimental)
+------------
+
+Original decommissioning algorithm has issues when DataNodes having lots of
```
Comment (goiri) [inline hadoop-hdfs-project/hadoop-hdfs/src/site/markdown/HdfsDataNodeAdminGuide.md]: "*The* original decommissioning algorithm"

File: hadoop-hdfs-project/hadoop-hdfs/src/site/markdown/HdfsDataNodeAdminGuide.md [hunk 3]
```diff
@@ -148,6 +148,35 @@ dfs.namenode.decommission.blocks.per.interval
...
+
+[HDFS-14854](https://issues.apache.org/jira/browse/HDFS-14854) introduced
+new decommission monitor in order to migigates those issues.
+This feature is currently marked as experimental and disabled by default.
+You can enable this by setting the value of
+`dfs.namenode.decommission.monitor.class` to
+`org.apache.hadoop.hdfs.server.blockmanagement.DatanodeAdminBackoffMonitor`
+in hdfs-site.xml.
+
+Relevant configuration properties are listed in the table below.
```
Comment (goiri) [inline hadoop-hdfs-project/hadoop-hdfs/src/site/markdown/HdfsDataNodeAdminGuide.md]: "*The* relevant"

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  markdownlint  |   0m  0s |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  18m  2s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   1m 19s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  31m 57s |  branch has no errors when building and testing our client artifacts.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m  2s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   1m  4s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  13m  8s |  patch has no errors when building and testing our client artifacts.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 32s |  The patch does not generate ASF License warnings.  |
|  |   |  50m  1s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1755/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1755 |
| Optional Tests | dupname asflicense mvnsite markdownlint |
| uname | Linux a5f77dcc62cf 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / b56c08b |
| Max. process+thread count | 419 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs U: hadoop-hdfs-project/hadoop-hdfs |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1755/2/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by iwasakims** [comment]
Thanks for the review, @goiri .

**Pull request merged**